accel-pppd/libnetlink/iputils.c

```diff
@@ -458,7 +463,78 @@ int __export ipaddr_del_peer(int ifindex, in_addr_t addr, in_addr_t peer)
 	return r;
 }
 
-int __export iproute_add(int ifindex, in_addr_t src, in_addr_t dst, in_addr_t gw, int proto, int mask, uint32_t prio)
+#ifdef HAVE_VRF
+static uint32_t ipvrf_get_table(const char *vrf_name)
+{
+	struct iplink_req {
+		struct nlmsghdr n;
+		struct ifinfomsg i;
+		char buf[4096];
+	} req;
+	struct rtnl_handle *rth = net->rtnl_get();
+	struct rtattr *tb[IFLA_MAX+1];
+	struct rtattr *li[IFLA_INFO_MAX+1];
+	struct rtattr *vrf_attr[IFLA_VRF_MAX + 1];
+	struct ifinfomsg *ifi;
+	int len;
+	uint32_t tb_id = RT_TABLE_MAIN;
+
+	log_ppp_info2("utils: getting route table for %s\n", vrf_name);
+
+	if (!vrf_name)
+		return tb_id;
```

Comment:
```suggestion
		return RT_TABLE_MAIN;
```

This would be more obvious I think.

Reply:
Now it returns 0 - "no error", so calls ipvrf_get_table(&id, NULL) should return the main table ID.